src/test/features/loading/status-announcer-spec.ts

```diff
@@ -0,0 +1,171 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {LoadingMixin} from '../../../features/loading.js';
+import {FINISHED_LOADING_ANNOUNCEMENT, INITIAL_STATUS_ANNOUNCEMENT, LoadingStatusAnnouncer, UPDATE_STATUS_DEBOUNCE_MS} from '../../../features/loading/status-announcer.js';
+import ModelViewerElementBase from '../../../model-viewer-base.js';
+import {assetPath, isInDocumentTree, timePasses, until, waitForEvent} from '../../helpers.js';
+
+const expect = chai.expect;
+const TIME_TO_UPDATE_MS = UPDATE_STATUS_DEBOUNCE_MS + 50;
+
+suite('LoadingStatusAnnouncer', () => {
+  let nextId = 0;
+  let tagName: string;
+  let ModelViewerElement: any;
+  let loadingStatusAnnouncer: LoadingStatusAnnouncer;
+
+  setup(() => {
+    tagName = `model-viewer-loading-announcer-${nextId++}`;
+    ModelViewerElement = class extends LoadingMixin
+    (ModelViewerElementBase) {
+      static get is() {
+        return tagName;
+      }
+    };
+    customElements.define(tagName, ModelViewerElement);
+    loadingStatusAnnouncer = new LoadingStatusAnnouncer();
+  });
+
+  suite('when a model is registered', () => {
+    suite('that has a src', () => {
+      test('sets initial status', async () => {
+        const element = new ModelViewerElement();
+        element.poster = assetPath('Astronaut.png');
+        element.src = assetPath('Astronaut.glb');
+
+        loadingStatusAnnouncer.registerInstance(element);
+
+        await timePasses(TIME_TO_UPDATE_MS);
+
+        const {statusElement} = loadingStatusAnnouncer;
+
+        expect(statusElement.textContent)
+            .to.be.equal(INITIAL_STATUS_ANNOUNCEMENT);
+      });
+
+      suite('after the model loads', () => {
+        test('sets the finished status', async () => {
+          const element = new ModelViewerElement();
+          element.src = assetPath('Astronaut.glb');
+
+          loadingStatusAnnouncer.registerInstance(element);
+
+          await until(() => element.loaded);
+          await timePasses(TIME_TO_UPDATE_MS);
+
+          const {statusElement} = loadingStatusAnnouncer;
+
+          expect(statusElement.textContent)
+              .to.be.equal(FINISHED_LOADING_ANNOUNCEMENT);
+        });
+      });
+
+      suite('there are other registered models', () => {
+        test('sets finished status when all models are loaded', async () => {
+          const elementOne = new ModelViewerElement();
+          const elementTwo = new ModelViewerElement();
+
+          elementOne.src = elementTwo.src = assetPath('Astronaut.glb');
+
+          loadingStatusAnnouncer.registerInstance(elementOne);
+          loadingStatusAnnouncer.registerInstance(elementTwo);
+
+          await Promise.all(
+              [until(() => elementOne.loaded), until(() => elementTwo.loaded)]);
+          await timePasses(TIME_TO_UPDATE_MS);
```

Comment:
This seems a bit fragile; similar to some of the async environment work, I wonder if we can expose some internal events for the debounce starting/stopping, similar to `Model`'s `envmap-change` event, which is really only for tests.

Reply:
I'll investigate this

Reply:
`LoadingStatusAnnouncer` now inherits from `EventDispatcher`. The tests have been updated to listen to events rather than wait for a fixed amount of time.